packages/rum-core/src/domain/requestCollection.ts

```diff
@@ -148,3 +158,23 @@ function getNextRequestIndex() {
   nextRequestIndex += 1
   return result
 }
+
+function waitForResponseToFinish(context: RumFetchCompleteContext, callback: (duration: Duration) => void) {
+  if (context.response && isExperimentalFeatureEnabled('fetch_duration')) {
+    const responseClone = context.response.clone()
+    if (responseClone.body) {
+      readBytesFromStream(
+        responseClone.body,
+        () => {
+          callback(elapsed(context.startClocks.timeStamp, timeStampNow()))
+        },
+        {
+          bytesLimit: Number.POSITIVE_INFINITY,
+          collectStreamBody: false,
+        }
+      )
+      return
+    }
+  }
+  callback(elapsed(context.startClocks.timeStamp, timeStampNow()))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏would it worth having tests for null response/body behaviors?